### PR TITLE
Sagas with multiple starting messages should be correlated properly

### PR DESCRIPTION
**Pull request by Scooletz**
This is a backport of  https://github.com/Particular/NServiceBus.AzureStoragePersistence/pull/32

Connects to https://github.com/Particular/PlatformDevelopment/issues/634

### CODE REVIEW

**Comment by Scooletz** [comment]
Ping @Particular/azure-maintainers This is a `cherry-picked` version of the Particular/NServiceBus.AzureStoragePersistence#32 ported back to the `Azure` repo. I'd love to add the tests for the storage but there's no infrastructure for this in this repo. Should I spend some time porting it as well?

**Comment by yvesgoeleven** [comment]
@Scooletz porting tests over would be best of course, is it doable?

**Comment by Scooletz** [comment]
@yvesgoeleven I can take a look at that and give at least some high lvl estimation. The only question is: 'Should I port it all' ;-)

**Comment by yvesgoeleven** [comment]
@Scooletz you can make that call, I have no view of what is entailed in these tests

File: packaging/nuget/nservicebus.azure.nuspec [hunk 1]
```diff
@@ -18,7 +18,7 @@
       <dependency id="Microsoft.Data.OData" version="[5.6.0.0, 6.0.0.0)" />
       <dependency id="Microsoft.Data.Services.Client" version="[5.6.0.0, 6.0.0.0)" />
       <dependency id="Microsoft.WindowsAzure.ConfigurationManager" version="[2.0.1.0, 3.0.0.0)" />
-      <dependency id="NServiceBus" version="[5.1.2.0, 6.0.0.0)" />
+      <dependency id="NServiceBus" version="[5.2.8.0, 6.0.0.0)" />
```
Comment (SimonCropp) [inline packaging/nuget/nservicebus.azure.nuspec]: why is a core update required? is this nuget using new features added in 5.2.8?

**Pull request closed without merging**